src/AzureClient/AzureClient.cs

```diff
@@ -262,13 +320,18 @@ public async Task<ExecutionResult> GetConnectionStatusAsync(IChannel channel)
             }
 
             var stream = new MemoryStream();
-            var protocol = await new JobStorageHelper(ConnectionString).DownloadJobOutputAsync(jobId, stream);
+            await new JobStorageHelper(ConnectionString).DownloadJobOutputAsync(jobId, stream);
             stream.Seek(0, SeekOrigin.Begin);
-            var outputJson = new StreamReader(stream).ReadToEnd();
+            var output = new StreamReader(stream).ReadToEnd();
+            var deserializedOutput = JsonConvert.DeserializeObject<Dictionary<string, JToken>>(output);
+            var histogram = new Dictionary<string, double>();
+            foreach (var entry in deserializedOutput["histogram"] as JObject)
```

Comment:
In other places, I've seen histograms serialized as lists of alternating keys and values where keys are lists of ints, since JSON doesn't support complex mapping keys, but this looks like it uses a string serialization of keys instead? Is that specific to some targets, or is there a conversion happening somewhere? Thanks for the help!

Reply:
From what I can tell, these histograms are always serialized in this way, where the keys are strings and the values are floats. @anpaz-msft, do you know if there is a better/more robust way to extract the histogram from the job output than what I am doing here? I don't see anything in the job definition that tells me how the output is serialized. There is the `OutputDataFormat` property, but that is just a string.

Reply:
@cesarzc, thanks for reviewing this PR! One last question for you: Do you have any opinion about the histogram deserialization discussed here?

Reply:
This is expected to be consistent across all providers. In this case, the serialization is done the way @cgranade described it, as a list of alternating keys and values.

A job output is a serialized bond data structures. In this case the specific data structure would be `QSharpExecutionOutput` and it is part of the `Microsoft.Quantum.Providers.Shared.Datastructures.v1` namespace.

Using bond would be the preferred way to deserialize from the stream. The code to do it would look something like the following:
```csharp
var simpleJsonReader = new SimpleJsonReader(stream);
var deserializedOutput = Deserialize<QSharpExecutionOutput>.From(simpleJsonReader);
```

In case I missed something, you can find more details about Bond here: https://microsoft.github.io/bond/manual/bond_cs.html

Reply:
Thanks @cesarzc! But actually that's not the format I get here, since I'm getting the result from storage. `JobStorageHelper.DownloadJobOutputAsync` gives me a stream that looks something like:
```
{"duration":10758761,"histogram":{"0":0.2499999999999999,"1":0.2499999999999999,"2":0.2499999999999999,"3":0.2499999999999999}}
```
Is it expected that this format is different than the `QSharpExecutionOutput` that you describe? Is it safe to assume that this also would be consistent across all providers?

Reply:
That's interesting, is this a QIO or Honeywell job?

I just checked an IonQ job, and its output was the following:
```
{"Histogram":["[0,0]",0.4999999999999999,"[1,1]",0.4999999999999999]}
```

Reply:
Mine is also an IonQ job. I think the difference is whether the job output comes directly from `IQuantumMachine.ExecuteAsync`, which is what you're showing, or whether the result is retrieved later from storage, which is what I'm showing.

Reply:
Followed up with @cesarzc offline. It turns out that there is a small bug in qsharp-runtime (at [this line](https://github.com/microsoft/qsharp-runtime/blob/c8f23017800974814f19d8ef005fb9786cd734bb/src/Azure/Azure.Quantum.Client/Storage/JobStorageHelper.cs#L122)) which is causing me to get the `"rawOutputData"` instead of the translated `"outputData"` for the job. I should indeed be getting the format @cgranade mentioned, and should be able to deserialize it as @cesarzc described.

I'll send a new PR for consuming that fix and updating the deserialization code here, once it's available.

Reply:
I've opened https://github.com/microsoft/qsharp-runtime/issues/239 to track this.